### ISSUE TITLE
Workflow fails because some dependencies can't be parsed as a gradle package

### ISSUE THREAD

**Issue by kevalpatel2106** [mikepenz/gradle-dependency-submission#218]
I started getting this error once I integrated the compose into the app. Compose uses bom in the project.

Failed workflow: https://github.com/kevalpatel2106/pocket-ci/actions/runs/4257717106/jobs/7408160892

Version catalogue: https://github.com/kevalpatel2106/pocket-ci/blob/a609ba05baacaf929440717ee1d65ef0bc24676a/gradle/libs.versions.toml

Log:
```
Error: The given 'androidx.compose.ui:ui-tooling FAILED can't be parsed as a gradle package.
[37](https://github.com/kevalpatel2106/pocket-ci/actions/runs/4257717106/jobs/7408160892#step:3:40)
      at parseGradlePackage (/home/runner/work/_actions/mikepenz/gradle-dependency-submission/v0.8.4/webpack:/gradle-dependency-submission/lib/parse.js:73:1)
[38](https://github.com/kevalpatel2106/pocket-ci/actions/runs/4257717106/jobs/7408160892#step:3:41)
      at parseGradleDependency (/home/runner/work/_actions/mikepenz/gradle-dependency-submission/v0.8.4/webpack:/gradle-dependency-submission/lib/parse.js:176:1)
[39](https://github.com/kevalpatel2106/pocket-ci/actions/runs/4257717106/jobs/7408160892#step:3:42)
      at parseGradleGraph (/home/runner/work/_actions/mikepenz/gradle-dependency-submission/v0.8.4/webpack:/gradle-dependency-submission/lib/parse.js:127:1)
[40](https://github.com/kevalpatel2106/pocket-ci/actions/runs/4257717106/jobs/7408160892#step:3:43)
      at /home/runner/work/_actions/mikepenz/gradle-dependency-submission/v0.8.4/webpack:/gradle-dependency-submission/lib/process.js:195:1
[41](https://github.com/kevalpatel2106/pocket-ci/actions/runs/4257717106/jobs/7408160892#step:3:44)
      at Generator.next (<anonymous>)
[42](https://github.com/kevalpatel2106/pocket-ci/actions/runs/4257717106/jobs/7408160892#step:3:45)
      at fulfilled (/home/runner/work/_actions/mikepenz/gradle-dependency-submission/v0.8.4/webpack:/gradle-dependency-submission/lib/process.js:28:1)
[43](https://github.com/kevalpatel2106/pocket-ci/actions/runs/4257717106/jobs/7408160892#step:3:46)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Output of compoes-ui-tooling dependency by running `./gradlew :app:depenedencies`

```
+--- androidx.compose.ui:ui-tooling -> 1.3.3
|    +--- androidx.annotation:annotation:1.1.0 -> 1.5.0 (*)
|    +--- androidx.compose.runtime:runtime:1.1.1 -> 1.3.3
|    |    \--- org.jetbrains.kotlinx:kotlinx-coroutines-android:1.6.4 (*)
|    +--- androidx.compose.ui:ui:1.3.3
|    |    +--- androidx.annotation:annotation:1.5.0 (*)
|    |    +--- androidx.compose.runtime:runtime-saveable:1.3.3
|    |    |    +--- androidx.annotation:annotation:1.1.0 -> 1.5.0 (*)
|    |    |    \--- androidx.compose.runtime:runtime:1.3.3 (*)
|    |    +--- androidx.compose.ui:ui-geometry:1.3.3
|    |    |    \--- androidx.annotation:annotation:1.1.0 -> 1.5.0 (*)
|    |    +--- androidx.compose.ui:ui-graphics:1.3.3
|    |    |    +--- androidx.annotation:annotation:1.2.0 -> 1.5.0 (*)
|    |    |    \--- androidx.compose.ui:ui-unit:1.3.3
|    |    |         +--- androidx.annotation:annotation:1.1.0 -> 1.5.0 (*)
|    |    |         \--- androidx.compose.ui:ui-geometry:1.3.3 (*)
|    |    +--- androidx.compose.ui:ui-text:1.3.3
|    |    |    +--- androidx.annotation:annotation:1.1.0 -> 1.5.0 (*)
|    |    |    +--- androidx.compose.ui:ui-graphics:1.3.3 (*)
|    |    |    \--- androidx.compose.ui:ui-unit:1.3.3 (*)
|    |    \--- androidx.compose.ui:ui-unit:1.3.3 (*)
|    +--- androidx.compose.ui:ui-tooling-data:1.3.3
|    |    +--- androidx.annotation:annotation:1.1.0 -> 1.5.0 (*)
|    |    +--- androidx.compose.runtime:runtime:1.2.0 -> 1.3.3 (*)
|    |    \--- androidx.compose.ui:ui:1.3.3 (*)
|    \--- androidx.compose.ui:ui-tooling-preview:1.3.3
|         +--- androidx.annotation:annotation:1.2.0 -> 1.5.0 (*)
|         \--- androidx.compose.runtime:runtime:1.1.1 -> 1.3.3 (*)
```

**Comment by mikepenz** [comment]
The log you are seeing logs the exact line we get from the comman: 
https://github.com/mikepenz/gradle-dependency-submission/blob/main/src/parse.ts#L45


So this log: `Error: The given 'androidx.compose.ui:ui-tooling FAILED can't be parsed as a gradle package.`

Means that something outputs `androidx.compose.ui:ui-tooling FAILED` in your build. 

Could it be that you use some dependency which can't be resolved?

**Comment by kevalpatel2106** [comment]
I cannot see any dependency resolution failing locally. I tried to print output of gradle dependencies on CI [here](https://github.com/kevalpatel2106/pocket-ci/actions/runs/4261364524/jobs/7415598000). But `androidx.compose.ui:ui-tooling` was not there on the list.

<img src="https://user-images.githubusercontent.com/20060162/221167708-d781f7a7-87c6-4040-8574-90b101c06762.png" width=450/>

Wdyt about adding a flag into the action to print the output of `gradle dependencies` command in the action if configured?

**Comment by kevalpatel2106** [comment]
Tried to add debug logs in https://github.com/mikepenz/gradle-dependency-submission/pull/220

**Comment by kevalpatel2106** [comment]
Ok, ran the job again in debug mode and looks like one of the task `debugImplementationDependenciesMetadata` is returning `androidx.compose.ui:ui-tooling FAILED`.

I wonder why the script is not picking up the version from the previous tasks where it successfully identified `androidx.compose.ui:ui-tooling -> 1.3.3`?

[Action](https://github.com/kevalpatel2106/pocket-ci/actions/runs/4257717106/jobs/7423971182)

**Comment by mikepenz** [comment]
That's interesting. While I can't say why gradle doesn't pick it up. 

It may be a good idea from the action to now fail in this case, but only log an error, instead of fully failing. 🤔

**Comment by mikepenz** [comment]
Could you please try the following PR and see if this would help to not fail the action at least if one of the dependencies can't be parsed:

```
mikepenz/gradle-dependency-submission@b976085384ae0f4b51a12d484238cc7fcde2c8a3
```

https://github.com/mikepenz/gradle-dependency-submission/pull/221

**Comment by kevalpatel2106** [comment]
> Could you please try the following PR and see if this would help to not fail the action at least if one of the dependencies can't be parsed:
> 
> ```
> mikepenz/gradle-dependency-submission@7e91645a2e8b9aae09e618b85369fb21e2aaf36d
> ```
> 
> #221

Sure, I will try and get back to you.




> That's interesting. While I can't say why gradle doesn't pick it up.
> 
> It may be a good idea from the action to now fail in this case, but only log an error, instead of fully failing. 🤔

FWIW, when I removed Compose BOM and replaced the compose dependencies with a specific version, the workflow started working again and there were no failed dependency resolutions.

**Comment by mikepenz** [comment]
Released as v0.8.5
